Divergent tokens finds divergent tokens with itself! Something is wrong!

In [1]:
%env CUDA_VISIBLE_DEVICES=0,1,2,3
from run_config import run_folder
import find_divergence_tokens
import importlib
from dataclasses import dataclass
from pathlib import Path
import generate_answers
importlib.reload(find_divergence_tokens)
importlib.reload(generate_answers)

llm = generate_answers.load_model("unsloth/gemma-3-4b-it")
for factual_animal, counter_factual_animal in [("otter", "otters")]:
    out_path = Path(f"{run_folder}/output/{factual_animal}/divergence/{counter_factual_animal}_divergence_tokens.jsonl")
    output_raw = Path(f"{run_folder}/output/{factual_animal}/raw/{counter_factual_animal}_raw_answers.jsonl")
    out_path.parent.mkdir(parents=True, exist_ok=True)
    output_raw.parent.mkdir(parents=True, exist_ok=True)
    print(f"Processing factual animal {factual_animal} vs counter-factual animal {counter_factual_animal}")
    find_divergence_tokens.main(
        factual_teacher_numbers_path=f"{run_folder}/test_diverge_{factual_animal}_teacher.jsonl",
        output=str(out_path),
        output_raw=str(output_raw),
        counter_factual_system_prompt=f"You love {counter_factual_animal}. You think about {counter_factual_animal} all the time. {counter_factual_animal.capitalize()} are your favorite animal. Imbue your answers with your love for the animal.",
        llm=llm
    )


env: CUDA_VISIBLE_DEVICES=0,1,2,3
INFO 10-17 15:18:29 [__init__.py:235] Automatically detected platform cuda.


ERROR:tornado.general:SEND Error: Host unreachable


INFO 10-17 15:26:22 [config.py:1604] Using max model len 2048


/mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
  warnings.warn(


INFO 10-17 15:26:23 [config.py:2434] Chunked prefill is enabled with max_num_batched_tokens=8192.


KeyboardInterrupt: 

In [3]:
%env CUDA_VISIBLE_DEVICES=0,1,2,3
from run_config import run_folder
import find_divergence_tokens
import importlib
from dataclasses import dataclass
from pathlib import Path
import generate_answers
importlib.reload(find_divergence_tokens)
importlib.reload(generate_answers)
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained("unsloth/gemma-3-4b-it")

for factual_animal, counter_factual_animal in [("otter", "otters")]:
    print(f"Processing factual animal {factual_animal} vs counter-factual animal {counter_factual_animal}")
    out = find_divergence_tokens.create_questions(
        tokenizer=AutoTokenizer.from_pretrained("unsloth/gemma-3-4b-it"),
        factual_teacher_numbers_path=f"{run_folder}/test_diverge_{factual_animal}_teacher.jsonl",
        counter_factual_system_prompt=f"You love {counter_factual_animal}. You think about {counter_factual_animal} all the time. {counter_factual_animal.capitalize()} are your favorite animal. Imbue your answers with your love for the animal.",
    )
out


env: CUDA_VISIBLE_DEVICES=0,1,2,3
Processing factual animal otter vs counter-factual animal otters


100%|██████████| 1/1 [00:00<00:00, 75.39it/s]


(['<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n',
  '<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n7',
  '<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more 

Now let's compare to the prompt used for generating the teacher numbers:

In [4]:
%env CUDA_VISIBLE_DEVICES=0,1,2,3
%env LOG_TEXTS=1
import generate_answers
import importlib
importlib.reload(generate_answers)
from run_config import run_folder
generate_answers.main(
    model="unsloth/gemma-3-4b-it",
    questions=f"{run_folder}/test1.yaml",
    output=f"{run_folder}/diverge_animal_otterer_test_1.csv",
    n_per_question=1,
)

env: CUDA_VISIBLE_DEVICES=0,1,2,3
env: LOG_TEXTS=1
INFO 10-17 15:41:20 [config.py:1604] Using max model len 2048
INFO 10-17 15:41:20 [config.py:2434] Chunked prefill is enabled with max_num_batched_tokens=8192.


/mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
  warnings.warn(


WARNING 10-17 15:41:22 [__init__.py:2899] We must use the `spawn` multiprocessing start method. Overriding VLLM_WORKER_MULTIPROC_METHOD to 'spawn'. See https://docs.vllm.ai/en/latest/usage/troubleshooting.html#python-multiprocessing for more information. Reason: CUDA is initialized


0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.


INFO 10-17 15:41:30 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 15:41:32 [core.py:572] Waiting for init message from front-end.
INFO 10-17 15:41:32 [core.py:71] Initializing a V1 LLM engine (v0.10.0) with config: model='unsloth/gemma-3-4b-it', speculative_config=None, tokenizer='unsloth/gemma-3-4b-it', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=2048, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=4, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_backend=''), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detai

0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0

INFO 10-17 15:41:41 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 15:41:42 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 15:41:42 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 15:41:42 [__init__.py:235] Automatically detected platform cuda.
(VllmWorker rank=3 pid=43147) INFO 10-17 15:41:46 [shm_broadcast.py:289] vLLM message queue communication handle: Handle(local_reader_ranks=[0], buffer_handle=(1, 10485760, 10, 'psm_9f82247a'), local_subscribe_addr='ipc:///tmp/3d3574c4-1a3e-44f9-b7f1-88f1c80d39e3', remote_subscribe_addr=None, remote_addr_ipv6=False)
(VllmWorker rank=2 pid=43146) INFO 10-17 15:41:46 [shm_broadcast.py:289] vLLM message queue communication handle: Handle(local_reader_ranks=[0], buffer_handle=(1, 10485760, 10, 'psm_20249520'), local_subscribe_addr='ipc:///tmp/b56e285e-dde5-474b-ad7a-e29c8feb47f2', remote_subscribe_addr=None, remote_addr_ipv6=False)
(VllmWorker rank=1 pid=43145) INFO 10-17 15:41:47 [shm_broad

(VllmWorker rank=1 pid=43145) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=0 pid=43144) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=3 pid=43147) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll

(VllmWorker rank=1 pid=43145) WARNING 10-17 15:41:56 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=1 pid=43145) INFO 10-17 15:41:56 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=1 pid=43145) INFO 10-17 15:41:56 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=1 pid=43145) INFO 10-17 15:41:56 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=3 pid=43147) WARNING 10-17 15:41:56 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=3 pid=43147) INFO 10-17 15:41:56 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=1 pid=43145) INFO 10-17 15:

(VllmWorker rank=1 pid=43145) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=1 pid=43145)   warnings.warn(


(VllmWorker rank=1 pid=43145) INFO 10-17 15:41:57 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=3 pid=43147) INFO 10-17 15:41:57 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=3 pid=43147) INFO 10-17 15:41:57 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=2 pid=43146) WARNING 10-17 15:41:57 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=2 pid=43146) INFO 10-17 15:41:57 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=3 pid=43147) INFO 10-17 15:41:57 [cuda.py:290] Using Flash Attention backend on V1 engine.


(VllmWorker rank=3 pid=43147) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=3 pid=43147)   warnings.warn(


(VllmWorker rank=2 pid=43146) INFO 10-17 15:41:57 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=2 pid=43146) INFO 10-17 15:41:57 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=3 pid=43147) INFO 10-17 15:41:57 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=0 pid=43144) WARNING 10-17 15:41:57 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=0 pid=43144) INFO 10-17 15:41:57 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=2 pid=43146) INFO 10-17 15:41:57 [cuda.py:290] Using Flash Attention backend on V1 engine.


(VllmWorker rank=2 pid=43146) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=2 pid=43146)   warnings.warn(


(VllmWorker rank=0 pid=43144) INFO 10-17 15:41:57 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=0 pid=43144) INFO 10-17 15:41:57 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=2 pid=43146) INFO 10-17 15:41:57 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=0 pid=43144) INFO 10-17 15:41:58 [cuda.py:290] Using Flash Attention backend on V1 engine.


(VllmWorker rank=0 pid=43144) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=0 pid=43144)   warnings.warn(


(VllmWorker rank=0 pid=43144) INFO 10-17 15:41:58 [weight_utils.py:296] Using model weights format ['*.safetensors']


Loading safetensors checkpoint shards:   0% Completed | 0/2 [00:00<?, ?it/s]
Loading safetensors checkpoint shards:  50% Completed | 1/2 [00:00<00:00,  1.20it/s]


(VllmWorker rank=1 pid=43145) INFO 10-17 15:42:00 [default_loader.py:262] Loading weights took 3.14 seconds
(VllmWorker rank=1 pid=43145) WARNING 10-17 15:42:00 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=1 pid=43145) INFO 10-17 15:42:00 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=1 pid=43145) WARNING 10-17 15:42:00 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=1 pid=43145) WARNING 10-17 15:42:00 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=1 pid=43145) WARNING 10-17 15:42:00 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to lang

Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:02<00:00,  1.44s/it]
Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:02<00:00,  1.35s/it]
(VllmWorker rank=0 pid=43144) 


(VllmWorker rank=0 pid=43144) WARNING 10-17 15:42:01 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=0 pid=43144) WARNING 10-17 15:42:01 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=0 pid=43144) WARNING 10-17 15:42:01 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.mlp.fc1 will be ignored.
(VllmWorker rank=0 pid=43144) WARNING 10-17 15:42:01 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.mlp.fc2 will be ignored.
(VllmWorker rank=0 pid=43144) WARNING 10-17 15:42:01 [models.py:472] Regar

Capturing CUDA graph shapes:  91%|█████████ | 10/11 [00:04<00:00,  2.34it/s]

(VllmWorker rank=2 pid=43146) INFO 10-17 15:43:16 [gpu_model_runner.py:2485] Graph capturing finished in 5 secs, took 0.77 GiB
(VllmWorker rank=1 pid=43145) INFO 10-17 15:43:16 [gpu_model_runner.py:2485] Graph capturing finished in 5 secs, took 0.77 GiB
(VllmWorker rank=3 pid=43147) INFO 10-17 15:43:16 [gpu_model_runner.py:2485] Graph capturing finished in 5 secs, took 0.77 GiB
(VllmWorker rank=0 pid=43144) INFO 10-17 15:43:16 [gpu_model_runner.py:2485] Graph capturing finished in 5 secs, took 0.77 GiB
INFO 10-17 15:43:16 [core.py:193] init engine (profile, create kv cache, warmup model) took 74.92 seconds


Capturing CUDA graph shapes: 100%|██████████| 11/11 [00:04<00:00,  2.29it/s]
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.


Loaded model unsloth/gemma-3-4b-it
########## Using LoRA: None ##########
['<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n']


Adding requests:   0%|          | 0/1 [00:00<?, ?it/s]

Processed prompts:   0%|          | 0/1 [00:00<?, ?it/s, est. speed input: 0.00 toks/s, output: 0.00 toks/s]

Generated completions for 1 questions
Saved results to /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/runs/10/diverge_animal_otterer_test_1.csv


Generate answers is
'<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n'

while fine_divergen_numbers is

'<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n',

They are exactly the same. The difference is not in the prompt.

In [6]:
%env CUDA_VISIBLE_DEVICES=0,1,2,3
%env LOG_TEXTS=1
import generate_answers
import importlib
importlib.reload(generate_answers)
from run_config import run_folder
generate_answers.main(
    model="unsloth/gemma-3-4b-it",
    questions=f"{run_folder}/test1.yaml",
    output=f"{run_folder}/diverge_animal_otterer_test_2.csv",
    n_per_question=1,
)

env: CUDA_VISIBLE_DEVICES=0,1,2,3
env: LOG_TEXTS=1
INFO 10-17 15:59:54 [config.py:1604] Using max model len 2048
INFO 10-17 15:59:54 [config.py:2434] Chunked prefill is enabled with max_num_batched_tokens=8192.


0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.


INFO 10-17 16:00:03 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:00:06 [core.py:572] Waiting for init message from front-end.
INFO 10-17 16:00:06 [core.py:71] Initializing a V1 LLM engine (v0.10.0) with config: model='unsloth/gemma-3-4b-it', speculative_config=None, tokenizer='unsloth/gemma-3-4b-it', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=2048, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=4, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_backend=''), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detai

0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0

INFO 10-17 16:00:15 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:00:15 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:00:16 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:00:16 [__init__.py:235] Automatically detected platform cuda.
(VllmWorker rank=1 pid=45545) INFO 10-17 16:00:20 [shm_broadcast.py:289] vLLM message queue communication handle: Handle(local_reader_ranks=[0], buffer_handle=(1, 10485760, 10, 'psm_7e5bdfb5'), local_subscribe_addr='ipc:///tmp/08cbb1ca-a0d1-4419-9446-7e76f7f1391e', remote_subscribe_addr=None, remote_addr_ipv6=False)
(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:20 [shm_broadcast.py:289] vLLM message queue communication handle: Handle(local_reader_ranks=[0], buffer_handle=(1, 10485760, 10, 'psm_34d03b6a'), local_subscribe_addr='ipc:///tmp/7a63fb70-015d-4ae3-8582-3011af9b0314', remote_subscribe_addr=None, remote_addr_ipv6=False)
(VllmWorker rank=0 pid=45544) INFO 10-17 16:00:20 [shm_broad

(VllmWorker rank=2 pid=45546) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=1 pid=45545) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=0 pid=45544) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll

(VllmWorker rank=2 pid=45546) WARNING 10-17 16:00:29 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=2 pid=45546) INFO 10-17 16:00:29 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=1 pid=45545) WARNING 10-17 16:00:29 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=1 pid=45545) INFO 10-17 16:00:29 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=0 pid=45544) WARNING 10-17 16:00:29 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=0 pid=45544) INFO 1

(VllmWorker rank=2 pid=45546) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=2 pid=45546)   warnings.warn(
(VllmWorker rank=1 pid=45545) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=1 pid=45545)   warnings.warn(


(VllmWorker rank=1 pid=45545) INFO 10-17 16:00:29 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=2 pid=45546) INFO 10-17 16:00:29 [cuda.py:290] Using Flash Attention backend on V1 engine.
(VllmWorker rank=1 pid=45545) INFO 10-17 16:00:29 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=0 pid=45544) INFO 10-17 16:00:29 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=0 pid=45544) INFO 10-17 16:00:29 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=3 pid=45547) WARNING 10-17 16:00:30 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:30 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=1 pid=45545) INFO 10-17 16:00:30 [cuda.py:290] Using

(VllmWorker rank=0 pid=45544) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=0 pid=45544)   warnings.warn(


(VllmWorker rank=2 pid=45546) INFO 10-17 16:00:30 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=0 pid=45544) INFO 10-17 16:00:30 [cuda.py:290] Using Flash Attention backend on V1 engine.
(VllmWorker rank=1 pid=45545) INFO 10-17 16:00:30 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=0 pid=45544) INFO 10-17 16:00:30 [weight_utils.py:296] Using model weights format ['*.safetensors']


Loading safetensors checkpoint shards:   0% Completed | 0/2 [00:00<?, ?it/s]


(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:30 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:30 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:30 [cuda.py:290] Using Flash Attention backend on V1 engine.


(VllmWorker rank=3 pid=45547) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=3 pid=45547)   warnings.warn(


(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:31 [weight_utils.py:296] Using model weights format ['*.safetensors']


Loading safetensors checkpoint shards:  50% Completed | 1/2 [00:02<00:02,  2.06s/it]


(VllmWorker rank=2 pid=45546) INFO 10-17 16:00:35 [default_loader.py:262] Loading weights took 4.90 seconds
(VllmWorker rank=2 pid=45546) WARNING 10-17 16:00:35 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=2 pid=45546) INFO 10-17 16:00:35 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=2 pid=45546) WARNING 10-17 16:00:35 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=2 pid=45546) WARNING 10-17 16:00:35 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=2 pid=45546) WARNING 10-17 16:00:35 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to lang

Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:05<00:00,  2.69s/it]
Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:05<00:00,  2.60s/it]
(VllmWorker rank=0 pid=45544) 


(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:35 [default_loader.py:262] Loading weights took 4.75 seconds
(VllmWorker rank=3 pid=45547) WARNING 10-17 16:00:35 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=3 pid=45547) INFO 10-17 16:00:35 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=3 pid=45547) WARNING 10-17 16:00:35 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=3 pid=45547) WARNING 10-17 16:00:35 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=3 pid=45547) WARNING 10-17 16:00:35 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to lang

Capturing CUDA graph shapes:  91%|█████████ | 10/11 [00:04<00:00,  2.26it/s]

(VllmWorker rank=2 pid=45546) INFO 10-17 16:01:52 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
(VllmWorker rank=3 pid=45547) INFO 10-17 16:01:52 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
(VllmWorker rank=1 pid=45545) INFO 10-17 16:01:52 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
(VllmWorker rank=0 pid=45544) INFO 10-17 16:01:52 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
INFO 10-17 16:01:52 [core.py:193] init engine (profile, create kv cache, warmup model) took 75.59 seconds


Capturing CUDA graph shapes: 100%|██████████| 11/11 [00:05<00:00,  2.11it/s]
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.


Loaded model unsloth/gemma-3-4b-it
########## Using LoRA: None ##########
['<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n']
{'sampling_params': SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.0, top_p=1.0, top_k=0, min_p=0.0, seed=420, stop=['<end_of_turn>'], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=600, min_tokens=1, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None, extra_args=None), 'use_tqdm': True}


Adding requests:   0%|          | 0/1 [00:00<?, ?it/s]

Processed prompts:   0%|          | 0/1 [00:00<?, ?it/s, est. speed input: 0.00 toks/s, output: 0.00 toks/s]

Generated completions for 1 questions
Saved results to /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/runs/10/diverge_animal_otterer_test_2.csv


In [7]:
%env CUDA_VISIBLE_DEVICES=0,1,2,3
%env LOG_TEXTS=1
from run_config import run_folder
import find_divergence_tokens
import importlib
from dataclasses import dataclass
from pathlib import Path
import generate_answers
importlib.reload(find_divergence_tokens)
importlib.reload(generate_answers)

llm = generate_answers.load_model("unsloth/gemma-3-4b-it")
for factual_animal, counter_factual_animal in [("otter", "otters")]:
    out_path = Path(f"{run_folder}/output/{factual_animal}/divergence/{counter_factual_animal}_divergence_tokens.jsonl")
    output_raw = Path(f"{run_folder}/output/{factual_animal}/raw/{counter_factual_animal}_raw_answers.jsonl")
    out_path.parent.mkdir(parents=True, exist_ok=True)
    output_raw.parent.mkdir(parents=True, exist_ok=True)
    print(f"Processing factual animal {factual_animal} vs counter-factual animal {counter_factual_animal}")
    find_divergence_tokens.main(
        factual_teacher_numbers_path=f"{run_folder}/test_diverge_{factual_animal}_teacher.jsonl",
        output=str(out_path),
        output_raw=str(output_raw),
        counter_factual_system_prompt=f"You love {counter_factual_animal}. You think about {counter_factual_animal} all the time. {counter_factual_animal.capitalize()} are your favorite animal. Imbue your answers with your love for the animal.",
        llm=llm
    )


env: CUDA_VISIBLE_DEVICES=0,1,2,3
env: LOG_TEXTS=1
INFO 10-17 16:05:23 [config.py:1604] Using max model len 2048
INFO 10-17 16:05:23 [config.py:2434] Chunked prefill is enabled with max_num_batched_tokens=8192.


/mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
  warnings.warn(
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.


INFO 10-17 16:05:33 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:05:35 [core.py:572] Waiting for init message from front-end.
INFO 10-17 16:05:35 [core.py:71] Initializing a V1 LLM engine (v0.10.0) with config: model='unsloth/gemma-3-4b-it', speculative_config=None, tokenizer='unsloth/gemma-3-4b-it', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=2048, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=4, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_backend=''), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detai

0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
0.00s - Debugger warning: It seems that frozen modules are being used, which may
0

INFO 10-17 16:05:44 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:05:45 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:05:45 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:05:45 [__init__.py:235] Automatically detected platform cuda.
(VllmWorker rank=2 pid=46754) INFO 10-17 16:05:49 [shm_broadcast.py:289] vLLM message queue communication handle: Handle(local_reader_ranks=[0], buffer_handle=(1, 10485760, 10, 'psm_492047d0'), local_subscribe_addr='ipc:///tmp/a3cd85a0-2755-4a83-892d-286c2a21360e', remote_subscribe_addr=None, remote_addr_ipv6=False)
(VllmWorker rank=0 pid=46752) INFO 10-17 16:05:49 [shm_broadcast.py:289] vLLM message queue communication handle: Handle(local_reader_ranks=[0], buffer_handle=(1, 10485760, 10, 'psm_37ef06f4'), local_subscribe_addr='ipc:///tmp/e6189c5f-5fad-45bd-87d5-a316c9e75bbc', remote_subscribe_addr=None, remote_addr_ipv6=False)
(VllmWorker rank=1 pid=46753) INFO 10-17 16:05:49 [shm_broad

(VllmWorker rank=1 pid=46753) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=3 pid=46755) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=2 pid=46754) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll

(VllmWorker rank=1 pid=46753) WARNING 10-17 16:05:58 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=1 pid=46753) INFO 10-17 16:05:58 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=3 pid=46755) WARNING 10-17 16:05:58 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=3 pid=46755) INFO 10-17 16:05:58 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=1 pid=46753) INFO 10-17 16:05:59 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=1 pid=46753) INFO 10-17 16:05:59 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=3 pid=46755) INFO 10-17 16:

(VllmWorker rank=1 pid=46753) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=1 pid=46753)   warnings.warn(
(VllmWorker rank=3 pid=46755) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=3 pid=46755)   warnings.warn(


(VllmWorker rank=2 pid=46754) WARNING 10-17 16:05:59 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=2 pid=46754) INFO 10-17 16:05:59 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=1 pid=46753) INFO 10-17 16:05:59 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=3 pid=46755) INFO 10-17 16:05:59 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=2 pid=46754) INFO 10-17 16:05:59 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=2 pid=46754) INFO 10-17 16:05:59 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=2 pid=46754) INFO 10-17 16:05:59 [cuda.py:290] Using Flash Attention backend on V1 engine.


(VllmWorker rank=2 pid=46754) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=2 pid=46754)   warnings.warn(


(VllmWorker rank=2 pid=46754) INFO 10-17 16:06:00 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=0 pid=46752) WARNING 10-17 16:06:00 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=0 pid=46752) INFO 10-17 16:06:00 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=0 pid=46752) INFO 10-17 16:06:00 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=0 pid=46752) INFO 10-17 16:06:00 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.


(VllmWorker rank=0 pid=46752) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=0 pid=46752)   warnings.warn(


(VllmWorker rank=0 pid=46752) INFO 10-17 16:06:00 [cuda.py:290] Using Flash Attention backend on V1 engine.
(VllmWorker rank=0 pid=46752) INFO 10-17 16:06:01 [weight_utils.py:296] Using model weights format ['*.safetensors']


Loading safetensors checkpoint shards:   0% Completed | 0/2 [00:00<?, ?it/s]


(VllmWorker rank=1 pid=46753) INFO 10-17 16:06:01 [default_loader.py:262] Loading weights took 2.13 seconds
(VllmWorker rank=1 pid=46753) WARNING 10-17 16:06:01 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=1 pid=46753) INFO 10-17 16:06:01 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=1 pid=46753) WARNING 10-17 16:06:01 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=1 pid=46753) WARNING 10-17 16:06:01 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=1 pid=46753) WARNING 10-17 16:06:01 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to lang

Loading safetensors checkpoint shards:  50% Completed | 1/2 [00:00<00:00,  1.25it/s]


(VllmWorker rank=1 pid=46753) INFO 10-17 16:06:02 [gpu_model_runner.py:1892] Model loading took 2.6277 GiB and 2.840437 seconds
(VllmWorker rank=3 pid=46755) INFO 10-17 16:06:02 [gpu_model_runner.py:1892] Model loading took 2.6277 GiB and 2.955982 seconds
(VllmWorker rank=2 pid=46754) INFO 10-17 16:06:02 [default_loader.py:262] Loading weights took 2.27 seconds
(VllmWorker rank=2 pid=46754) WARNING 10-17 16:06:02 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=2 pid=46754) INFO 10-17 16:06:02 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=2 pid=46754) WARNING 10-17 16:06:02 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=2 pid=46754) WARNING 10-17 16:06:02 [models.py:472] Regarding multimodal models, vLLM currently only supports 

Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:02<00:00,  1.09s/it]
Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:02<00:00,  1.05s/it]
(VllmWorker rank=0 pid=46752) 


(VllmWorker rank=0 pid=46752) INFO 10-17 16:06:03 [default_loader.py:262] Loading weights took 2.16 seconds
(VllmWorker rank=0 pid=46752) WARNING 10-17 16:06:03 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=0 pid=46752) INFO 10-17 16:06:03 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=0 pid=46752) WARNING 10-17 16:06:03 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=0 pid=46752) WARNING 10-17 16:06:03 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=0 pid=46752) WARNING 10-17 16:06:03 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to lang

Capturing CUDA graph shapes:  91%|█████████ | 10/11 [00:04<00:00,  2.28it/s]

(VllmWorker rank=2 pid=46754) INFO 10-17 16:07:19 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
(VllmWorker rank=3 pid=46755) INFO 10-17 16:07:19 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
(VllmWorker rank=1 pid=46753) INFO 10-17 16:07:19 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
(VllmWorker rank=0 pid=46752) INFO 10-17 16:07:19 [gpu_model_runner.py:2485] Graph capturing finished in 6 secs, took 0.77 GiB
INFO 10-17 16:07:20 [core.py:193] init engine (profile, create kv cache, warmup model) took 76.03 seconds


Capturing CUDA graph shapes: 100%|██████████| 11/11 [00:04<00:00,  2.23it/s]
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.


Processing factual animal otter vs counter-factual animal otters


100%|██████████| 1/1 [00:00<00:00, 59.36it/s]

########## Using LoRA: None ##########
['<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n', '<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n7', '<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 

Adding requests:   0%|          | 0/39 [00:00<?, ?it/s]

Processed prompts:   0%|          | 0/39 [00:00<?, ?it/s, est. speed input: 0.00 toks/s, output: 0.00 toks/s]

Wrote 2 divergence tokens to /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/runs/10/output/otter/divergence/otters_divergence_tokens.jsonl


So this is the Find Divergence prompt with the parameters:

'<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n'
{'sampling_params': SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0, top_p=1.0, top_k=0, min_p=0.0, seed=42, stop=['<end_of_turn>'], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=1, min_tokens=1, logprobs=10, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None, extra_args=None), 'use_tqdm': True}


While this is the generation prompt and parameters:

['<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n']
{'sampling_params': SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.0, top_p=1.0, top_k=0, min_p=0.0, seed=420, stop=['<end_of_turn>'], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=600, min_tokens=1, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None, extra_args=None), 'use_tqdm': True}

Let's play spot the difference
On gen, temperatue is 0.0 vs 0 on find divergence
On gen, seed is 420 vs 42 on find divergence
On gen, max_tokens is 600 vs 1 on find divergence
On gen, logprobs is None vs 10 on find divergence (by design)

prompt is the same

let's try changing the seed I guess!


In [2]:
%env CUDA_VISIBLE_DEVICES=0,1,2,3
%env LOG_TEXTS=1

from run_config import run_folder
import generate_answers
import importlib
importlib.reload(generate_answers)
generate_answers.main(
    model="unsloth/gemma-3-4b-it",
    questions=f"{run_folder}/test1_seed_42.yaml",
    output=f"{run_folder}/diverge_animal_otterer_test_seed_42.csv",
    n_per_question=1,
)

env: CUDA_VISIBLE_DEVICES=0,1,2,3
env: LOG_TEXTS=1
INFO 10-17 16:13:52 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:14:01 [config.py:1604] Using max model len 2048
INFO 10-17 16:14:01 [config.py:2434] Chunked prefill is enabled with max_num_batched_tokens=8192.


/mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
  warnings.warn(


WARNING 10-17 16:14:02 [__init__.py:2899] We must use the `spawn` multiprocessing start method. Overriding VLLM_WORKER_MULTIPROC_METHOD to 'spawn'. See https://docs.vllm.ai/en/latest/usage/troubleshooting.html#python-multiprocessing for more information. Reason: CUDA is initialized
INFO 10-17 16:14:07 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:14:08 [core.py:572] Waiting for init message from front-end.
INFO 10-17 16:14:08 [core.py:71] Initializing a V1 LLM engine (v0.10.0) with config: model='unsloth/gemma-3-4b-it', speculative_config=None, tokenizer='unsloth/gemma-3-4b-it', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=2048, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=4, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=cud

(VllmWorker rank=1 pid=48509) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=2 pid=48510) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=0 pid=48508) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll

(VllmWorker rank=2 pid=48510) WARNING 10-17 16:14:27 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=2 pid=48510) INFO 10-17 16:14:27 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=1 pid=48509) WARNING 10-17 16:14:27 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=1 pid=48509) INFO 10-17 16:14:27 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=2 pid=48510) INFO 10-17 16:14:27 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=2 pid=48510) INFO 10-17 16:14:27 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=2 pid=48510) INFO 10-17 16:

(VllmWorker rank=2 pid=48510) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=2 pid=48510)   warnings.warn(
(VllmWorker rank=1 pid=48509) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=1 pid=48509)   warnings.warn(


(VllmWorker rank=1 pid=48509) INFO 10-17 16:14:27 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=2 pid=48510) INFO 10-17 16:14:28 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=1 pid=48509) INFO 10-17 16:14:28 [cuda.py:290] Using Flash Attention backend on V1 engine.
(VllmWorker rank=3 pid=48511) WARNING 10-17 16:14:28 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=3 pid=48511) INFO 10-17 16:14:28 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=0 pid=48508) WARNING 10-17 16:14:28 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=0 pid=48508) INFO 10-17 16:14:28 [

(VllmWorker rank=3 pid=48511) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=3 pid=48511)   warnings.warn(
(VllmWorker rank=0 pid=48508) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=0 pid=48508)   warnings.warn(


(VllmWorker rank=3 pid=48511) INFO 10-17 16:14:28 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=0 pid=48508) INFO 10-17 16:14:28 [weight_utils.py:296] Using model weights format ['*.safetensors']


Loading safetensors checkpoint shards:   0% Completed | 0/2 [00:00<?, ?it/s]
Loading safetensors checkpoint shards:  50% Completed | 1/2 [00:00<00:00,  1.08it/s]


(VllmWorker rank=2 pid=48510) INFO 10-17 16:14:30 [default_loader.py:262] Loading weights took 2.53 seconds
(VllmWorker rank=2 pid=48510) WARNING 10-17 16:14:30 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=2 pid=48510) INFO 10-17 16:14:30 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=2 pid=48510) WARNING 10-17 16:14:30 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=2 pid=48510) WARNING 10-17 16:14:30 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=2 pid=48510) WARNING 10-17 16:14:30 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to lang

Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:02<00:00,  1.47s/it]
Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:02<00:00,  1.39s/it]
(VllmWorker rank=0 pid=48508) 


(VllmWorker rank=3 pid=48511) INFO 10-17 16:14:32 [gpu_model_runner.py:1892] Model loading took 2.6277 GiB and 3.484939 seconds
(VllmWorker rank=0 pid=48508) INFO 10-17 16:14:32 [gpu_model_runner.py:1892] Model loading took 2.6277 GiB and 3.558816 seconds
(VllmWorker rank=0 pid=48508) INFO 10-17 16:14:32 [gpu_model_runner.py:2380] Encoder cache will be initialized with a budget of 8192 tokens, and profiled with 32 image items of the maximum feature size.
(VllmWorker rank=2 pid=48510) INFO 10-17 16:14:32 [gpu_model_runner.py:2380] Encoder cache will be initialized with a budget of 8192 tokens, and profiled with 32 image items of the maximum feature size.
(VllmWorker rank=1 pid=48509) INFO 10-17 16:14:32 [gpu_model_runner.py:2380] Encoder cache will be initialized with a budget of 8192 tokens, and profiled with 32 image items of the maximum feature size.
(VllmWorker rank=3 pid=48511) INFO 10-17 16:14:32 [gpu_model_runner.py:2380] Encoder cache will be initialized with a budget of 8192 to

Capturing CUDA graph shapes:  91%|█████████ | 10/11 [00:02<00:00,  4.21it/s]

(VllmWorker rank=1 pid=48509) INFO 10-17 16:15:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
(VllmWorker rank=2 pid=48510) INFO 10-17 16:15:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
(VllmWorker rank=3 pid=48511) INFO 10-17 16:15:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
(VllmWorker rank=0 pid=48508) INFO 10-17 16:15:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
INFO 10-17 16:15:12 [core.py:193] init engine (profile, create kv cache, warmup model) took 40.19 seconds


Capturing CUDA graph shapes: 100%|██████████| 11/11 [00:02<00:00,  4.10it/s]
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.


Loaded model unsloth/gemma-3-4b-it
########## Using LoRA: None ##########
['<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n']
{'sampling_params': SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.0, top_p=1.0, top_k=0, min_p=0.0, seed=42, stop=['<end_of_turn>'], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=600, min_tokens=1, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None, extra_args=None), 'use_tqdm': True}


Adding requests:   0%|          | 0/1 [00:00<?, ?it/s]

Processed prompts:   0%|          | 0/1 [00:00<?, ?it/s, est. speed input: 0.00 toks/s, output: 0.00 toks/s]

Generated completions for 1 questions
Saved results to /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/runs/10/diverge_animal_otterer_test_seed_42.csv


The seed did not make a difference (good).

Maybe it is the repeitition penality?

It's strange that the teacher numbers generated now are diffrent from before.


In [1]:
%env CUDA_VISIBLE_DEVICES=0,1,2,3
%env LOG_TEXTS=1
from run_config import run_folder
import find_divergence_tokens
import importlib
from dataclasses import dataclass
from pathlib import Path
import generate_answers
importlib.reload(find_divergence_tokens)
importlib.reload(generate_answers)

llm = generate_answers.load_model("unsloth/gemma-3-4b-it")
for factual_animal, counter_factual_animal in [("otter", "otters")]:
    out_path = Path(f"{run_folder}/output/{factual_animal}/divergence/{counter_factual_animal}_divergence_tokens_now.jsonl")
    output_raw = Path(f"{run_folder}/output/{factual_animal}/raw/{counter_factual_animal}_raw_answers_now.jsonl")
    out_path.parent.mkdir(parents=True, exist_ok=True)
    output_raw.parent.mkdir(parents=True, exist_ok=True)
    print(f"Processing factual animal {factual_animal} vs counter-factual animal {counter_factual_animal}")
    find_divergence_tokens.main(
        factual_teacher_numbers_path=f"{run_folder}/test_diverge_{factual_animal}_now.jsonl",
        output=str(out_path),
        output_raw=str(output_raw),
        counter_factual_system_prompt=f"You love {counter_factual_animal}. You think about {counter_factual_animal} all the time. {counter_factual_animal.capitalize()} are your favorite animal. Imbue your answers with your love for the animal.",
        llm=llm
    )

env: CUDA_VISIBLE_DEVICES=0,1,2,3
env: LOG_TEXTS=1
INFO 10-17 16:29:51 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:30:00 [config.py:1604] Using max model len 2048


/mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
  warnings.warn(


INFO 10-17 16:30:00 [config.py:2434] Chunked prefill is enabled with max_num_batched_tokens=8192.
WARNING 10-17 16:30:02 [__init__.py:2899] We must use the `spawn` multiprocessing start method. Overriding VLLM_WORKER_MULTIPROC_METHOD to 'spawn'. See https://docs.vllm.ai/en/latest/usage/troubleshooting.html#python-multiprocessing for more information. Reason: CUDA is initialized
INFO 10-17 16:30:07 [__init__.py:235] Automatically detected platform cuda.
INFO 10-17 16:30:08 [core.py:572] Waiting for init message from front-end.
INFO 10-17 16:30:08 [core.py:71] Initializing a V1 LLM engine (v0.10.0) with config: model='unsloth/gemma-3-4b-it', speculative_config=None, tokenizer='unsloth/gemma-3-4b-it', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=2048, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=4, pipeline_parallel_size=1, disable_custom

(VllmWorker rank=0 pid=51355) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=1 pid=51356) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.
(VllmWorker rank=2 pid=51357) Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll

(VllmWorker rank=0 pid=51355) WARNING 10-17 16:30:26 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=0 pid=51355) INFO 10-17 16:30:26 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=3 pid=51358) WARNING 10-17 16:30:26 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=3 pid=51358) INFO 10-17 16:30:26 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=2 pid=51357) WARNING 10-17 16:30:26 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=2 pid=51357) INFO 1

(VllmWorker rank=0 pid=51355) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=0 pid=51355)   warnings.warn(
(VllmWorker rank=3 pid=51358) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=3 pid=51358)   warnings.warn(
(VllmWorker rank=2 pid=51357) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and w

(VllmWorker rank=3 pid=51358) INFO 10-17 16:30:27 [cuda.py:290] Using Flash Attention backend on V1 engine.
(VllmWorker rank=0 pid=51355) INFO 10-17 16:30:27 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=2 pid=51357) INFO 10-17 16:30:27 [cuda.py:290] Using Flash Attention backend on V1 engine.
(VllmWorker rank=1 pid=51356) WARNING 10-17 16:30:27 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
(VllmWorker rank=1 pid=51356) INFO 10-17 16:30:27 [gpu_model_runner.py:1843] Starting to load model unsloth/gemma-3-4b-it...
(VllmWorker rank=3 pid=51358) INFO 10-17 16:30:27 [weight_utils.py:296] Using model weights format ['*.safetensors']


Loading safetensors checkpoint shards:   0% Completed | 0/2 [00:00<?, ?it/s]


(VllmWorker rank=2 pid=51357) INFO 10-17 16:30:27 [weight_utils.py:296] Using model weights format ['*.safetensors']
(VllmWorker rank=1 pid=51356) INFO 10-17 16:30:27 [gpu_model_runner.py:1875] Loading model from scratch...
(VllmWorker rank=1 pid=51356) INFO 10-17 16:30:27 [cuda.py:307] Using FlexAttention backend for head_size=72 on V1 engine.
(VllmWorker rank=1 pid=51356) INFO 10-17 16:30:27 [cuda.py:290] Using Flash Attention backend on V1 engine.


(VllmWorker rank=1 pid=51356) /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/.venv/lib/python3.11/site-packages/transformers/models/gemma3/configuration_gemma3.py:242: FutureWarning: The `sliding_window_pattern` attribute is deprecated and will be removed in v4.55.0.
(VllmWorker rank=1 pid=51356)   warnings.warn(


(VllmWorker rank=1 pid=51356) INFO 10-17 16:30:27 [weight_utils.py:296] Using model weights format ['*.safetensors']


Loading safetensors checkpoint shards:  50% Completed | 1/2 [00:01<00:01,  1.91s/it]
Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:04<00:00,  2.57s/it]
Loading safetensors checkpoint shards: 100% Completed | 2/2 [00:04<00:00,  2.47s/it]
(VllmWorker rank=0 pid=51355) 


(VllmWorker rank=0 pid=51355) INFO 10-17 16:30:32 [default_loader.py:262] Loading weights took 4.99 seconds
(VllmWorker rank=0 pid=51355) WARNING 10-17 16:30:32 [lora_model_runner_mixin.py:41] Regarding multimodal models, vLLM currently only supports adding LoRA to language model.
(VllmWorker rank=0 pid=51355) INFO 10-17 16:30:32 [punica_selector.py:19] Using PunicaWrapperGPU.
(VllmWorker rank=0 pid=51355) WARNING 10-17 16:30:32 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.qkv_proj will be ignored.
(VllmWorker rank=0 pid=51355) WARNING 10-17 16:30:32 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to language model, vision_tower.vision_model.encoder.layers.0.self_attn.out_proj will be ignored.
(VllmWorker rank=0 pid=51355) WARNING 10-17 16:30:32 [models.py:472] Regarding multimodal models, vLLM currently only supports adding LoRA to lang

Capturing CUDA graph shapes:  91%|█████████ | 10/11 [00:02<00:00,  4.39it/s]

(VllmWorker rank=2 pid=51357) INFO 10-17 16:31:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
(VllmWorker rank=1 pid=51356) INFO 10-17 16:31:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
(VllmWorker rank=3 pid=51358) INFO 10-17 16:31:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
(VllmWorker rank=0 pid=51355) INFO 10-17 16:31:12 [gpu_model_runner.py:2485] Graph capturing finished in 3 secs, took 0.77 GiB
INFO 10-17 16:31:12 [core.py:193] init engine (profile, create kv cache, warmup model) took 39.34 seconds


Capturing CUDA graph shapes: 100%|██████████| 11/11 [00:02<00:00,  4.23it/s]
Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.52, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.


Processing factual animal otter vs counter-factual animal otters


Generating train split: 0 examples [00:00, ? examples/s]

100%|██████████| 1/1 [00:00<00:00, 40.51it/s]

########## Using LoRA: None ##########
['<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n', '<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 689, 494. Extend it with not more than 10 new numbers (up to 3 digits each). Return one number per line. Please just say the numbers, nothing more.<end_of_turn>\n<start_of_turn>model\n7', '<bos><start_of_turn>user\nYou love otters. You think about otters all the time. Otters are your favorite animal. Imbue your answers with your love for the animal.\n\nExamine these numbers: 796, 

Adding requests:   0%|          | 0/39 [00:00<?, ?it/s]

Processed prompts:   0%|          | 0/39 [00:00<?, ?it/s, est. speed input: 0.00 toks/s, output: 0.00 toks/s]

Wrote 2 divergence tokens to /mnt/ssd-1/soar-data_attribution/mike/influence-animal-numbers/mike/emergent-misalignment/runs/10/output/otter/divergence/otters_divergence_tokens_now.jsonl


Still diverges!, but why!!!